app/assets/javascripts/discourse/app/components/sidebar.js

```diff
@@ -1,3 +1,39 @@
 import GlimmerComponent from "discourse/components/glimmer";
+import { bind } from "discourse-common/utils/decorators";
 
-export default class Sidebar extends GlimmerComponent {}
+export default class Sidebar extends GlimmerComponent {
+  constructor() {
+    super(...arguments);
+
+    if (this.site.mobileView) {
+      document.addEventListener("click", this.collapseSidebar);
+    }
+  }
+
+  @bind
+  collapseSidebar(event) {
+    let shouldCollapseSidebar = false;
+
+    const isClickWithinSidebar = event.composedPath().some((element) => {
+      if (
+        element?.className !== "sidebar-section-header-caret" &&
+        ["A", "BUTTON"].includes(element.nodeName)
+      ) {
+        shouldCollapseSidebar = true;
+        return true;
+      }
```

Comment:
Instead of waiting for the `page:changed` event which I feel is too late, let's just try collapsing on links and button clicks within the sidebar.

Reply:
@CvX how do you feel about having https://github.com/zeppelin/ember-click-outside in core ?

Reply:
+1! I'd very much like to replace some of our custom code with a neat element modifier.